### ISSUE TITLE
Writing video to AVI does not work

### ISSUE THREAD

**Issue by choeger** [savonet/ocaml-mm#2]
When trying to write multiple images, the resulting file cannot be played. This seems to be caused by tools expecting every frame in one CHUNK. 

the write method should look like this:

```
      method write buf ofs len =
        for i = ofs to ofs + len - 1 do
          let s = Image.RGBA32.to_RGB24_string buf.(i) in
          self#output "00db";
          self#output_int (String.length s);
          self#output s;
          datalen <- datalen + String.length s + 8;
        done;
        dataframes <- dataframes + len
```

(close and initializer need to be adapted as well, I cannot make a patch just now)

**Comment by smimram** [comment]
Yes, the current code is not satisfactory, and should be rewritten quite a bit (in particular, we'd like to be able to write audio as well). The code we actually use is here: https://github.com/savonet/liquidsoap/blob/master/src/tools/avi.ml and this one was more of a demo. To make it short: we would be glad to accept patches, but probably won't have time to update this soon...
